engine/sql/core/src/main/scala/org/apache/spark/sql/index/TreapIndexedRelation.scala

```diff
@@ -52,10 +53,9 @@ private[sql] case class TreapIndexedRelation(
       val eval_key = BindReferences.bindReference(column_keys.head, child.output).eval(row)
       eval_key match {
         case key: Number => (key.doubleValue, row)
-        case key: GenericInternalRow =>
-          val row_array = key.values(0).asInstanceOf[GenericArrayData].array
-          require(row_array.length == 1)
-          (row_array.head.asInstanceOf[Double], row)
+        case key: Point =>
```

Comment:
The problem why you cannot do this is that you need a total order for all the elements in a one-dimensional index. Without spacing filling curves, you are not able to create such a total order for multi-dimensional elements.